src/services/authentication.rs

```diff
@@ -65,7 +65,7 @@ impl Service {
             .await
             .map_err(|_| ServiceError::InternalServerError)?;
 
-        verify_password(password.as_bytes(), &user_authentication)?;
```

Comment:
Just out of curiosity, why are you wrapping the `ServiceError::InvalidPassword` to a `ServiceError:WrongPasswordOrUsername`. Is it because you want to have the `Invalid password` error info in the validate_password function and then return a more generic error to the caller?

Reply:
Hi @josecelano, In the "change password" form, I want to inform the user that the current password is not valid. However, in the "login form," I don't want to reveal why the login failed, whether it was because of a wrong email or password. If we returned the "invalid password," people could use the login form to find out if a given email/username is registered in the application.